homeassistant/components/tado/config_flow.py

```diff
@@ -159,6 +160,56 @@ async def async_step_import(
             },
         )
 
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a reconfiguration flow initialized by the user."""
+        self.config_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reconfigure_confirm()
+
+    async def async_step_reconfigure_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a reconfiguration flow initialized by the user."""
+        errors: dict[str, str] = {}
+        assert self.config_entry
+
+        if user_input is not None:
+            user_input[CONF_USERNAME] = self.config_entry.data[CONF_USERNAME]
+            try:
+                await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except PyTado.exceptions.TadoWrongCredentialsException:
+                errors["base"] = "invalid_auth"
```

Comment:
Correct, in normal circumstances. This PR is mostly inspired by a previous [impactful error](https://github.com/home-assistant/core/issues/114517), caused by the underlying used PyTado library. This caused a failed setup, due to an issue with Tado with the summertime switch. This caused the the refresh token _not being properly refreshed_ and also not triggering an unauthorized error - odd yes, still it happened.
I see this PR as an extra service of control within HA to control mechanisms with Tado. Also, a normal update interval (if no device trackers are used) is five minutes. If users are impatient, they can instantly update the password instead of waiting for the re-auth issue appearing.

Reply:
Makes sense, thanks for clarification 👍